subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/device/adb/AdbDeviceParser.kt

```diff
@@ -0,0 +1,26 @@
+package com.avito.runner.service.worker.device.adb
+
+class AdbDeviceParser {
+
+    fun parse(output: String): Set<AdbDeviceParams> {
+        return output
+            .substringAfter("List of devices attached")
+            .split(System.lineSeparator())
+            .asSequence()
+            .map { it.trim() }
+            .filter { it.isNotEmpty() }
+            .filter { it.contains("online") || it.contains("device") }
+            .map {
+                AdbDeviceParams(
+                    id = it.substringBefore(" "),
+                    model = it.substringAfter("model:").substringBefore(" device"),
+                    online = when {
+                        it.contains("offline", ignoreCase = true) -> false
+                        it.contains("device", ignoreCase = true) -> true
+                        else -> throw IllegalStateException("Unknown devicesManager output for device: $it")
```

Comment:
how this exception will be handled on upper level? is it safe to throw it here, or it could ends up with hanging process?

Reply:
it have been presented in original code, so nothing changed